### PR TITLE
[NO-JIRA] Add suppliers and CPE for packages

### PR DESCRIPTION
**Pull request by surendrapathak**
[NO-JIRA] Add CPE for packages

1. blake2 is known as a vendor in CPE
2. tiny_sha3 follows a typically seen pattern of org-name:repo-name at github
3. mpdecimal is a wildcard as no previously registered CPE for bytereef or mpdecimal

[NO-JIRA] Add known suppliers for packages

Supplier is an NTIA Minimum Elements recommendation. For open-source, it is recommended to include the name of the organization supporting or person managing the project to be included as a supplier.

### CODE REVIEW

**Comment by surendrapathak** [comment]
It attempts to handle issues #6 and #7 - albeit without certainty.

File: build-sbom.py [hunk 1]
```diff
@@ -75,10 +77,12 @@ def is_match(pattern: str) -> bool:
         primary_package_purpose=spdx.PackagePurpose.SOURCE,
         download_url="https://www.bytereef.org/software/mpdecimal/releases/mpdecimal-2.5.1.tar.gz",
         download_hash_sha256="9f9cd4c041f99b5c49ffb7b59d9f12d95b683d88585608aa56a6307667b2b21f",
+        cpes=["cpe:2.3:a:bytereef:mpdecimal:2.5.1:*:*:*:*:*:*:*"],
```
Comment (sethmlarson) [inline build-sbom.py:80]: I don't like specifying CPEs that don't actually exist, it feels too hand-wavy to me, is there any guidance on what to do when a CPE doesn't exist?

Comment (surendrapathak) [reply build-sbom.py:80]: I don't disagree with the uneasiness here as this is a best-effort approach only and can be seen as useful or not. 

[Official CPE](https://cpe.mitre.org/specification/) has no exact coverage but some recommendations in this area. 

First, on finding products that do not have an official CPE:
> An organization may discover new products in use that do not yet have official identifiers; the organization could create identifiers, add them to its own extended dictionary, submit them for inclusion in the [Official CPE Dictionary](http://nvd.nist.gov/cpe.cfm), and use them internally while waiting for their addition to the Official CPE Dictionary.

Secondly, on extending and using the official dictionary:
> Organizations may create their own extended CPE dictionaries, which are used to store identifier names not present in the Official CPE Dictionary. There are several reasons why extended dictionaries are needed. For example, an organization may have to create identifier names for proprietary products that are only useful within that organization, such as internally developed software not found outside the organization. Another possible reason is that an IT company may want to use identifiers for their unreleased products that do not yet have official identifier names; once the identifier names have stabilized, the company would submit them to the Official CPE Dictionary.

And, finally, on the actual submission processing:
> CPE Names (also called CPE Identifiers, CPE-IDs, and CPEs) are created by the [CPE Community](https://cpe.mitre.org/community/discussion_list.html) on an as-needed basis, and must be well-formed as determined by the appropriate [CPE Specifications](https://cpe.mitre.org/specification/). Community members who require a new CPE Name should create a CPE-conformant name and submit it to the CPE Team for inclusion in the CPE Dictionary. The process is as follows:
>1. Refer to the appropriate CPE Specifications to develop your proposed CPE Name(s), making your best effort to create well-formed names. It is also recommended that you refer to the current [CPE Dictionary](http://nvd.nist.gov/cpe.cfm) for examples and precedents.
> 2. Prepare each CPE Name submission in valid CPE Dictionary XML form as defined by the [CPE 2.2 Dictionary Schema](https://cpe.mitre.org/specification/2.2/cpe-dictionary_2.2.xsd), which is also included by NIST on its CPE Dictionary page. You may bundle multiple name submissions into a single XML file. We recommend that you run an XML Schema validator to ensure your submission validates against the CPE Dictionary Schema.
> 3. Send your submission via email to [cpe_dictionary@nist.gov](mailto:cpe_dictionary@nist.gov).
> 4. NIST will conduct a quality assurance review before adding new names to the CPE Dictionary. During this review, portions of a proposed CPE Name(s) may be changed based on research performed against the most credible vendor and product information resources available. No name should be considered accepted until it is published in the CPE Dictionary.
>5. NIST will notify you when new CPE Names have been reviewed and approved.

So, creating and using the 'best-effort' CPE is not discouraged as long as it is based on "examples and precedents" (of course, it is ideal to merge "best-effort" CPE generation with its submission to NIST).

Returning to the PR, the included CPEs are 'best-effort' based on my own analysis of how CPEs have been assigned to both github and independent projects in the past. 

These CPEs are not misidentifying any component today by [vendor or product parts of the CPE](https://cpe.mitre.org/specification/). Therefore these provide a better than zero chance of mapping to any future vulnerability against this product. If the CPE is omitted from the SBOM, any future vulnerability is guaranteed not be mappable back to the SBOM via CPE.

The only theoretical downside is if the included CPE is indeed assigned to a _different_ product. 
That downside will produce false-positive against this SBOM for any future vulnerability in the other product. Obviously, there is no known vulnerability on that other product today or else the CPE already be official.

So, including the 'best-effort' CPE is essentially a trade-off between 'potential good vulnerability mapping' vs 'potential false positive'. I tend to lean towards surfacing the information.

**Review by sethmlarson — APPROVED** [review]
Thanks, I'll submit the CPEs that we've used to NVD to see if they'll accept them or give us official ones.

**Pull request merged**